rotkehlchen/chain/evm/decoding/stakedao/utils.py

```diff
@@ -20,36 +26,70 @@
 SUPPORTED_STAKEDAO_STRATEGIES: Final = {'curve', 'balancer', 'pendle', 'pancakeswap', 'angle'}
 
 
-def query_stakedao_gauges() -> None:
+def query_stakedao_gauges(evm_inquirer: 'EvmNodeInquirer') -> None:
     """Query StakeDAO gauges for lockers and strategies from the API and cache them."""
-    all_gauges: dict[ChainID, set] = defaultdict(set)
-
+    all_gauges: set[tuple[ChecksumEvmAddress, ChecksumEvmAddress]] = set()
+    chain_id = evm_inquirer.chain_id.serialize()
     try:  # Query lockers
         response_data = request_get_dict(url=f'{STAKEDAO_API}/lockers')
-        for item in response_data['parsed']:
+        for item in response_data.get('parsed', []):
+            if item['chainId'] != chain_id:
+                continue
+
             try:
-                all_gauges[ChainID.deserialize(item['chainId'])].add(deserialize_evm_address(item['modules']['gauge']))
+                all_gauges.add((
+                    deserialize_evm_address(item['modules']['gauge']),
+                    deserialize_evm_address(item['sdToken']['address']),  # underlying token
+                ))
             except (KeyError, DeserializationError) as e:
                 log.warning(f'Skipping stakedao locker entry {item} due to {e!s}')
                 continue
```

Comment:
here the KeyError should have what we do about the error message in other palkces